### PR TITLE
Change `new T()` to `GG::Wnd::Create<T>()` in GG and UI code.

### PR DESCRIPTION
**Pull request by LGM-Doyle**
This will make converting these lines to use `shared_ptr` only require changing the return type of `GG::Wnd::Create<T>` from `T*` to `shared_ptr<T>`.

### CODE REVIEW

File: UI/CUIControls.cpp [hunk 1]
```diff
@@ -1419,10 +1431,10 @@ SpeciesSelector::SpeciesSelector(GG::X w, GG::Y h) :
     Resize(GG::Pt(w, h - 8));
     const SpeciesManager& sm = GetSpeciesManager();
     for (SpeciesManager::playable_iterator it = sm.playable_begin(); it != sm.playable_end(); ++it)
-        Insert(new SpeciesRow(it->second, w, h - 4));
+        Insert(GG::Wnd::Create<SpeciesRow>(it->second, w, h - 4));
     if (!this->Empty()) {
         // Add an option for random selection
-        Insert(new SpeciesRow("RANDOM", UserString("GSETUP_RANDOM"), UserString("GSETUP_SPECIES_RANDOM_DESC"), w, h - 4,
+        Insert(GG::Wnd::Create<SpeciesRow>("RANDOM", UserString("GSETUP_RANDOM"), UserString("GSETUP_SPECIES_RANDOM_DESC"), w, h - 4,
                               ClientUI::GetTexture(ClientUI::ArtDir() / "icons/unknown.png")));
```
Comment (geoffthemedio) [inline UI/CUIControls.cpp]: indenting

File: UI/CUIControls.cpp [hunk 2]
```diff
@@ -1300,7 +1312,7 @@ void StatisticIcon::DoLayout() {
 
     if (!m_text) {
         // Create new label in correct place.
-        m_text = new CUILabel(text_elements.Text(), text_elements.Elements(),
+        m_text = GG::Wnd::Create<CUILabel>(text_elements.Text(), text_elements.Elements(),
                               format, GG::NO_WND_FLAGS,
```
Comment (geoffthemedio) [inline UI/CUIControls.cpp]: indent

**Comment by geoffthemedio** [comment]
Don't want to scan through the whole thing, but please fix indenting of constructor parameters after the new -> Wnd::Create substitution.

File: UI/MultiplayerLobbyWnd.cpp [hunk 3]
```diff
@@ -409,8 +409,8 @@ namespace {
 
             // host
             if (HumanClientApp::GetApp()->Networking().PlayerIsHost(player_id)) {
-                push_back(GG::Wnd::Create<GG::StaticGraphic>(GetHostTexture(),
-                    GG::GRAPHIC_CENTER | GG::GRAPHIC_FITGRAPHIC | GG::GRAPHIC_PROPSCALE, GG::INTERACTIVE));
+                push_back(GG::Wnd::Create<GG::StaticGraphic>(
+                              GetHostTexture(), GG::GRAPHIC_CENTER | GG::GRAPHIC_FITGRAPHIC | GG::GRAPHIC_PROPSCALE, GG::INTERACTIVE));
```
Comment (geoffthemedio) [inline UI/MultiplayerLobbyWnd.cpp]: ?

**Comment by LGM-Doyle** [comment]
Fixed the indenting.

File: UI/CUILinkTextBlock.cpp [hunk 4]
```diff
@@ -7,7 +7,7 @@ CUILinkTextBlock::CUILinkTextBlock(const std::string& str, const std::shared_ptr
                                    GG::Flags<GG::TextFormat> format, const GG::Clr& color,
                                    GG::Flags<GG::WndFlag> flags) :
     GG::BlockControl(GG::X0, GG::Y0, GG::X1, flags | GG::INTERACTIVE),
-    m_link_text(new CUILinkTextMultiEdit(str, GG::MULTI_WORDBREAK | GG::MULTI_READ_ONLY | GG::MULTI_LEFT |
+    m_link_text(GG::Wnd::Create<CUILinkTextMultiEdit>(str, GG::MULTI_WORDBREAK | GG::MULTI_READ_ONLY | GG::MULTI_LEFT |
                                               GG::MULTI_LINEWRAP | GG::MULTI_TOP | GG::MULTI_NO_HSCROLL |
```
Comment (geoffthemedio) [inline UI/CUILinkTextBlock.cpp]: ...

File: UI/ClientUI.cpp [hunk 5]
```diff
@@ -619,8 +619,8 @@ ClientUI::ClientUI() :
     m_player_list_wnd = new PlayerListWnd(PLAYER_LIST_WND_NAME);
     InitializeWindows();
 
-    m_intro_screen =            new IntroScreen();
-    m_multiplayer_lobby_wnd =   new MultiPlayerLobbyWnd();
+    m_intro_screen =            GG::Wnd::Create<IntroScreen>();
```
Comment (geoffthemedio) [inline UI/ClientUI.cpp]: should probably remove the extra spaces here if

File: UI/CombatReport/CombatReportWnd.cpp [hunk 6]
```diff
@@ -20,11 +20,13 @@ class CombatReportWnd::Impl {
...
-        m_graphical(new GraphicalSummaryWnd()),
-        m_log(new CombatLogWnd(m_wnd.ClientWidth(), m_wnd.ClientHeight())),
-        m_log_scroller(new GG::ScrollPanel(GG::X(0), GG::Y(0), m_tabs->ClientWidth(), m_tabs->ClientHeight(), m_log)),
+        m_tabs(GG::Wnd::Create<GG::TabWnd>(GG::X0, GG::Y0, GG::X1, GG::Y1, ClientUI::GetFont(),
+                                           ClientUI::CtrlColor(), ClientUI::TextColor())),
+        m_graphical(GG::Wnd::Create<GraphicalSummaryWnd>()),
+        m_log(GG::Wnd::Create<CombatLogWnd>(m_wnd.ClientWidth(), m_wnd.ClientHeight())),
+        m_log_scroller(
+            GG::Wnd::Create<GG::ScrollPanel>(
+                GG::X(0), GG::Y(0), m_tabs->ClientWidth(), m_tabs->ClientHeight(), m_log)),
```
Comment (geoffthemedio) [inline UI/CombatReport/CombatReportWnd.cpp]: could use `GG::X0` and `GG::Y0`

File: UI/CombatReport/GraphicalSummary.cpp [hunk 7]
```diff
@@ -352,18 +352,18 @@ class SideBar : public GG::Wnd {
 
         GG::Clr axis_label_color = combat_summary.SideColor();
 
-        m_x_axis_label = new CUILabel( (boost::format( UserString("COMBAT_FLEET_HEALTH_AXIS_LABEL") )
+        m_x_axis_label = GG::Wnd::Create<CUILabel>( (boost::format( UserString("COMBAT_FLEET_HEALTH_AXIS_LABEL") )
                                       % combat_summary.SideName()
```
Comment (geoffthemedio) [inline UI/CombatReport/GraphicalSummary.cpp]: odd indenting. could just revert to a single tab (4 spaces) indent from the previous line

File: UI/DesignWnd.cpp [hunk 8]
```diff
@@ -2558,16 +2561,19 @@ void SavedDesignsListBox::BaseRightClicked(GG::ListBox::iterator it, const GG::P
...
     // create popup menu with a commands in it
-    CUIPopupMenu popup(pt.x, pt.y);
+    auto popup = GG::Wnd::Create<CUIPopupMenu>(pt.x, pt.y);
     if (design->Producible())
-        popup.AddMenuItem(GG::MenuItem(UserString("DESIGN_ADD"),       false, false, add_design_action));
-    popup.AddMenuItem(GG::MenuItem(UserString("DESIGN_WND_DELETE_SAVED"), false, false, delete_saved_design_action));
-    popup.AddMenuItem(GG::MenuItem(UserString("DESIGN_WND_ADD_ALL_SAVED_NOW"),   false, false, add_all_saved_designs_action));
-    popup.AddMenuItem(GG::MenuItem(true)); // separator
-    popup.AddMenuItem(GG::MenuItem(UserString("DESIGN_WND_ADD_ALL_SAVED_START"), false, add_all,
+        popup->AddMenuItem(GG::MenuItem(UserString("DESIGN_ADD"),       false, false, add_design_action));
```
Comment (geoffthemedio) [inline UI/DesignWnd.cpp]: preferably align `false` or remove the extra spaces

File: UI/GalaxySetupWnd.cpp [hunk 9]
```diff
@@ -171,12 +171,15 @@ const GG::X GameRulesPanel::DefaultWidth() {
 
 GameRulesPanel::GameRulesPanel(GG::X w, GG::Y h) :
     GG::Control(GG::X0, GG::Y0, w, h, GG::NO_WND_FLAGS)
+{}
+
+void GameRulesPanel::CompleteConstruction()
 {
```
Comment (geoffthemedio) [inline UI/GalaxySetupWnd.cpp]: for single line function name / params, should put `{` on same line

File: UI/MultiplayerLobbyWnd.cpp [hunk 10]
```diff
@@ -227,34 +227,34 @@ namespace {
...
                 at(5)->SetMinSize(GG::Pt(GG::X(ClientUI::Pts()), PlayerFontHeight()));
                 at(5)->SetBrowseModeTime(GetOptionsDB().Get<int>("UI.tooltip-delay"));
                 at(5)->SetBrowseInfoWnd(std::make_shared<TextBrowseWnd>(
                     m_player_data.m_player_ready ? UserString("READY_BN") : UserString("NOT_READY_BN"),
                     "", PlayerReadyBrowseWidth()));
                 if (HumanClientApp::GetApp()->Networking().PlayerIsHost(player_id)) {
-                    push_back(new GG::StaticGraphic(GetHostTexture(),
-                        GG::GRAPHIC_CENTER | GG::GRAPHIC_FITGRAPHIC | GG::GRAPHIC_PROPSCALE, GG::INTERACTIVE));
+                    push_back(GG::Wnd::Create<GG::StaticGraphic>(
+                                  GetHostTexture(), GG::GRAPHIC_CENTER | GG::GRAPHIC_FITGRAPHIC | GG::GRAPHIC_PROPSCALE, GG::INTERACTIVE));
```
Comment (geoffthemedio) [inline UI/MultiplayerLobbyWnd.cpp]: ?

File: UI/MultiplayerLobbyWnd.cpp [hunk 11]
```diff
@@ -287,11 +287,11 @@ namespace {
...
             // ready state
             if (player_data.m_client_type == Networking::CLIENT_TYPE_AI_PLAYER) {
-                push_back(new CUILabel(""));
+                push_back(GG::Wnd::Create<CUILabel>(""));
                 at(5)->SetMinSize(GG::Pt(GG::X(ClientUI::Pts()), PlayerFontHeight()));
             } else {
-                push_back(new GG::StaticGraphic(GetReadyTexture(m_player_data.m_player_ready),
-                    GG::GRAPHIC_CENTER | GG::GRAPHIC_FITGRAPHIC | GG::GRAPHIC_PROPSCALE, GG::INTERACTIVE));
+                push_back(GG::Wnd::Create<GG::StaticGraphic>(
+                              GetReadyTexture(m_player_data.m_player_ready), GG::GRAPHIC_CENTER | GG::GRAPHIC_FITGRAPHIC | GG::GRAPHIC_PROPSCALE, GG::INTERACTIVE));
```
Comment (geoffthemedio) [inline UI/MultiplayerLobbyWnd.cpp]: ?

File: UI/MultiplayerLobbyWnd.cpp [hunk 12]
```diff
@@ -301,10 +301,10 @@ namespace {
 
             // host
             if (HumanClientApp::GetApp()->Networking().PlayerIsHost(player_id)) {
-                push_back(new GG::StaticGraphic(GetHostTexture(),
-                    GG::GRAPHIC_CENTER | GG::GRAPHIC_FITGRAPHIC | GG::GRAPHIC_PROPSCALE, GG::INTERACTIVE));
+                push_back(GG::Wnd::Create<GG::StaticGraphic>(
+                              GetHostTexture(), GG::GRAPHIC_CENTER | GG::GRAPHIC_FITGRAPHIC | GG::GRAPHIC_PROPSCALE, GG::INTERACTIVE));
```
Comment (geoffthemedio) [inline UI/MultiplayerLobbyWnd.cpp]: ?

File: UI/MultiplayerLobbyWnd.cpp [hunk 13]
```diff
@@ -409,10 +409,10 @@ namespace {
...
```
Comment (geoffthemedio) [inline UI/MultiplayerLobbyWnd.cpp]: ?

File: UI/ObjectListWnd.cpp [hunk 14]
```diff
@@ -1659,11 +1659,14 @@ class ObjectHeaderPanel : public GG::Control {
...
-        popup.AddMenuItem(std::move(planets_submenu));
-        popup.AddMenuItem(std::move(fleets_submenu));
+        popup->AddMenuItem(std::move(meters_submenu));
+        popup->AddMenuItem(std::move(planets_submenu));
+        popup->AddMenuItem(std::move(fleets_submenu));
 
-        popup.Run();
+        popup->Run();
+
+    // TODO remove when converting to shared_ptr
```
Comment (geoffthemedio) [inline UI/ObjectListWnd.cpp]: indent

File: UI/ProductionWnd.cpp [hunk 15]
```diff
@@ -702,9 +702,12 @@ namespace {
             if (UserStringExists(item_name))
                 item_name = UserString(item_name);
             std::string popup_label = boost::io::str(FlexibleFormat(UserString("ENC_LOOKUP")) % item_name);
-            popup.AddMenuItem(GG::MenuItem(popup_label, false, false, pedia_action));
+            popup->AddMenuItem(GG::MenuItem(popup_label, false, false, pedia_action));
 
-            popup.Run();
+            popup->Run();
+
+    // TODO remove when converting to shared_ptr
```
Comment (geoffthemedio) [inline UI/ProductionWnd.cpp]: indent

File: UI/ResearchWnd.cpp [hunk 16]
```diff
@@ -335,9 +335,12 @@ class ResearchQueueListBox : public QueueListBox {
             tech_name = UserString(queue_row->elem.name);
 
         std::string popup_label = boost::io::str(FlexibleFormat(UserString("ENC_LOOKUP")) % tech_name);
-        popup.AddMenuItem(GG::MenuItem(popup_label, false, false, pedia_action));
+        popup->AddMenuItem(GG::MenuItem(popup_label, false, false, pedia_action));
 
-        popup.Run();
+        popup->Run();
+
+    // TODO remove when converting to shared_ptr
```
Comment (geoffthemedio) [inline UI/ResearchWnd.cpp]: indent

File: UI/SidePanel.cpp [hunk 17]
```diff
@@ -820,9 +821,12 @@ class SidePanel::SystemNameDropDownList : public CUIDropDownList {
         };
 
         if (m_order_issuing_enabled && system->OwnedBy(HumanClientApp::GetApp()->EmpireID()))
-            popup.AddMenuItem(GG::MenuItem(UserString("SP_RENAME_SYSTEM"), false, false, rename_action));
+            popup->AddMenuItem(GG::MenuItem(UserString("SP_RENAME_SYSTEM"), false, false, rename_action));
 
-        popup.Run();
+        popup->Run();
+
+    // TODO remove when converting to shared_ptr
```
Comment (geoffthemedio) [inline UI/SidePanel.cpp]: indent

File: UI/SitRepPanel.cpp [hunk 18]
```diff
@@ -698,13 +702,19 @@ void SitRepPanel::DismissalMenu(GG::ListBox::iterator it, const GG::Pt& pt, cons
...
-    popup.AddMenuItem(GG::MenuItem(entry_margin + UserString("HOTKEY_COPY"), false, false, copy_action));
-    popup.AddMenuItem(GG::MenuItem(entry_margin + UserString("POPUP_MENU_PEDIA_PREFIX") +
+    popup->AddMenuItem(GG::MenuItem(entry_margin + UserString("HOTKEY_COPY"), false, false, copy_action));
+    popup->AddMenuItem(GG::MenuItem(entry_margin + UserString("POPUP_MENU_PEDIA_PREFIX") +
                                    UserString("SITREP_IGNORE_BLOCK_TITLE"),
                                    false, false, help_action));
 
-    if (!popup.Run())
+    if (!popup->Run())
+    {
```
Comment (geoffthemedio) [inline UI/SitRepPanel.cpp]: `{` to previous line

File: UI/SpecialsPanel.cpp [hunk 19]
```diff
@@ -142,12 +142,15 @@ bool SpecialsPanel::EventFilter(GG::Wnd* w, const GG::WndEvent& event) {
...
+        auto popup = GG::Wnd::Create<CUIPopupMenu>(pt.x, pt.y);
         std::string popup_label = boost::io::str(FlexibleFormat(UserString("ENC_LOOKUP")) % UserString(entry.first));
 
-        popup.AddMenuItem(GG::MenuItem(popup_label, false, false, zoom_action));
+        popup->AddMenuItem(GG::MenuItem(popup_label, false, false, zoom_action));
 
-        popup.Run();
+        popup->Run();
+
+    // TODO remove when converting to shared_ptr
```
Comment (geoffthemedio) [inline UI/SpecialsPanel.cpp]: indent

File: UI/ProductionWnd.cpp (hunk 15 above)
Comment (LGM-Doyle) [reply UI/ProductionWnd.cpp]: fixed.

File: UI/ResearchWnd.cpp (hunk 16 above)
Comment (LGM-Doyle) [reply UI/ResearchWnd.cpp]: fixed

File: UI/SidePanel.cpp (hunk 17 above)
Comment (LGM-Doyle) [reply UI/SidePanel.cpp]: fixed

File: UI/SitRepPanel.cpp (hunk 18 above)
Comment (LGM-Doyle) [reply UI/SitRepPanel.cpp]: fixed

File: UI/SpecialsPanel.cpp (hunk 19 above)
Comment (LGM-Doyle) [reply UI/SpecialsPanel.cpp]: fixed

**Review by LGM-Doyle — COMMENTED** [review]
All changes were made and it is rebased.

File: UI/DesignWnd.cpp [hunk 20]
```diff
@@ -2485,32 +2485,35 @@ void CompletedDesignsListBox::BaseRightClicked(GG::ListBox::iterator it, const G
...
 
     // save design
-    popup.AddMenuItem(GG::MenuItem(UserString("DESIGN_SAVE"), false, false, save_design_action));
+    popup->AddMenuItem(GG::MenuItem(UserString("DESIGN_SAVE"), false, false, save_design_action));
 
-    popup.AddMenuItem(GG::MenuItem(true)); // separator
-    popup.AddMenuItem(GG::MenuItem(UserString("DESIGN_WND_ADD_ALL_DEFAULT_START"), false, add_defaults,
+    popup->AddMenuItem(GG::MenuItem(true)); // separator
+    popup->AddMenuItem(GG::MenuItem(UserString("DESIGN_WND_ADD_ALL_DEFAULT_START"), false, add_defaults,
                                    toggle_add_default_designs_at_game_start_action));
```
Comment (geoffthemedio) [inline UI/DesignWnd.cpp:2511]: one space off...

Comment (LGM-Doyle) [reply UI/DesignWnd.cpp:2511]: fixed in master.

**Pull request merged**